conda/base/constants.py

```diff
@@ -24,25 +24,44 @@
 
 APP_NAME = 'conda'
 
-SEARCH_PATH = (
-    '/etc/conda/.condarc',
-    '/etc/conda/condarc',
-    '/etc/conda/condarc.d/',
-    '/var/lib/conda/.condarc',
-    '/var/lib/conda/condarc',
-    '/var/lib/conda/condarc.d/',
-    '$CONDA_ROOT/.condarc',
-    '$CONDA_ROOT/condarc',
-    '$CONDA_ROOT/condarc.d/',
-    '~/.conda/.condarc',
-    '~/.conda/condarc',
-    '~/.conda/condarc.d/',
-    '~/.condarc',
-    '$CONDA_PREFIX/.condarc',
-    '$CONDA_PREFIX/condarc',
-    '$CONDA_PREFIX/condarc.d/',
-    '$CONDARC',
-)
+if on_win:
+    SEARCH_PATH = (
+        '$CONDA_ROOT/.condarc',
+        '$CONDA_ROOT/condarc',
+        '$CONDA_ROOT/condarc.d/',
+        '~/.conda/.condarc',
+        '~/.conda/condarc',
+        '~/.conda/condarc.d/',
+        '~/.condarc',
+        '$CONDA_PREFIX/.condarc',
+        '$CONDA_PREFIX/condarc',
+        '$CONDA_PREFIX/condarc.d/',
+        '$CONDARC',
+        'C:/ProgramData/conda/.condarc',
+        'C:/ProgramData/conda/condarc',
+        'C:/ProgramData/conda/condarc.d'
```

Comment:
So, to get my system config (`C:/ProgramData/conda`) to take precedence I had to have it at the *bottom* of the `SEARCH_PATH` list and I'm not sure if that's the correct thing to do.

Reading the blog post again, it actually shows the precedence being `ENV --> USER --> SYSTEM`
```
# file: /etc/conda/condarc
channels: 
  - one 
  - two

# file: ~/.condarc
channels: 
  - three 
  - four

# file: $CONDA_PREFIX/.condarc
channels: 
  - five 
  - six
```
results in:
```
channels: 
  - five 
  - six 
  - three 
  - four 
  - one 
  - two
```
i.e. the system config should be the least significant rather than the most?

That being the case it seems `C:/ProgramData/conda` should be at the top of the list and that user-specified channels will take precedence to those specified at the system level.

Is that correct? Is that the desired default behaviour?

Reply:
As a sysadmin I would like user channels to be *appended* to the channel list - i.e for the channels I specify to take precedence in the config (but also able to be overridden by the user on the command line).

I'm fine changing the precedence to put `C:/ProgramData/conda` at the top / least significant position if that's the right thing to do - I think I can get my desired behaviour by using `#!top` in my system config